src/main/java/org/prebid/cache/repository/redis/RedisPropertyConfiguration.java

```diff
@@ -1,29 +1,104 @@
 package org.prebid.cache.repository.redis;
 
+import io.lettuce.core.RedisClient;
 import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.reactive.RedisStringReactiveCommands;
+import io.lettuce.core.cluster.RedisClusterClient;
+import io.lettuce.core.cluster.api.StatefulRedisClusterConnection;
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.requireNonNull;
+
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 @Configuration
 @ConditionalOnProperty(prefix = "spring.redis", name = {"host"})
 @ConfigurationProperties(prefix = "spring.redis")
 public class RedisPropertyConfiguration {
+
     private String host;
-    private Integer port;
+    private long timeout;
     private String password;
+    private Boolean isCluster;
+
+    private RedisURI createRedisURI() {
+        requireNonNull(host);
+        String[] hostParams = host.split(":");
+
+        if (isNull(password)) {
```

Comment:
What is the reason to not use old good fashioned `password != null`? `Objects.isNull()` was invented for use as method reference, for example in `Stream.filter()`, not as a general purpose replacement to `== null` comparison.